scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1493 @@
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``RandomState`` instance, then that
+        instance is used.
+
+    References
+    ----------
+    .. [1] Mckay et al., "A Comparison of Three Methods for Selecting Values
+       of Input Variables in the Analysis of Output from a Computer Code",
+       Technometrics, 1979.
+    .. [2] M. Stein, "Large sample properties of simulations using Latin
+       hypercube sampling." Technometrics 29, no. 2: 143-151, 1987.
+    .. [3] A. B. Owen, "Monte Carlo variance of scrambled net quadrature."
+       SIAM Journal on Numerical Analysis 34, no. 5: 1884-1910, 1997
+    .. [4]  Loh, W.-L. "On Latin hypercube sampling." The annals of statistics
+       24, no. 5: 2058-2080, 1996.
+
+    Examples
+    --------
+    Generate samples from a latin hypercube generator.
+
+    >>> from scipy.stats import qmc
+    >>> sampler = qmc.LatinHypercube(dim=2, seed=12345)
+    >>> sample = sampler.random(n_samples=5)
+    >>> sample
+    array([[0.01407678, 0.53672489],
+           [0.36321624, 0.75908794],
+           [0.28645499, 0.48089106],
+           [0.2070971 , 0.46936458],
+           [0.45021867, 0.66928603]])
+
+    Compute the quality of the sample using the discrepancy criterion.
+
+    >>> qmc.discrepancy(sample)
+    0.1271335273223828
+
+    Finally, samples can be scaled to bounds.
+
+    >>> bounds = [[0, 2], [10, 5]]
+    >>> qmc.scale(sample, bounds)
+    array([[0.14076781, 3.61017467],
+           [3.63216238, 4.27726383],
+           [2.86454994, 3.44267318],
+           [2.07097096, 3.40809374],
+           [4.50218672, 4.00785808]])
+
+    """
+
+    def __init__(self, dim, centered=False, seed=None):
+        super().__init__(dim=dim, seed=seed)
+        self.centered = centered
+
+    def random(self, n_samples=1):
+        """Draw `n_samples` in the half-open interval ``[0, 1)``.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples to generate in the parameter space.
+
+        Returns
+        -------
+        sample : array_like (n_samples, dim)
+            LHS sample.
+
+        """
+        if self.centered:
+            r = 0.5
+        else:
+            r = self.rng.random_sample((n_samples, self.dim))
+
+        q = self.rng.randint(low=1, high=n_samples,
+                             size=(n_samples, self.dim))
+
+        return 1. / n_samples * (q - r)
+
+
+class OptimalDesign(QMCEngine):
+    """Optimal design.
+
+    Optimize the design by doing random permutations to lower the centered
+    discrepancy. If `optimization` is False, `niter` design are generated and
+    the one with lowest centered discrepancy is return. This option is faster.
+
+    Centered discrepancy based design show better space filling robustness
+    toward 2D and 3D subprojections. Distance based design better space
+    filling but less robust to subprojections.
```

Comment:
Consider putting this sort of thing in a Notes section. Without familiarity with the subject, it is unintelligible.
```suggestion
    Centered discrepancy-based design shows better space filling robustness
    toward 2D and 3D subprojections. Distance-based design shows better space
    filling but less robustness to subprojections.
```
(Grammatical suggestions above may be incorrect because I don't fully understand what it's saying.)